src/NodeComposerPlugin.php

```diff
@@ -38,12 +38,26 @@ public function activate(Composer $composer, IOInterface $io)
         $this->io = $io;
 
         $extraConfig = $this->composer->getPackage()->getExtra();
+        $packageConfig = $extraConfig['mariusbuescher']['node-composer'] ?? [];
 
-        if (!isset($extraConfig['mariusbuescher']['node-composer'])) {
+        $this->updateConfigFromPackageProvidesConfig(
+            $packageConfig,
+            'node-version',
+            'imponeer/composer-nodejs-installer',
```

Comment:
Is this a hard coded package name? What will happen if you decide to delete this package?

I see a lot of problems coming from here.

Reply:
Yeah, it's hardcoded package name. If you remove it, node-composer will work as before this package was installed. It means that you need specify node version manually in your package config. So, probably if you use this package and specify also minimal version in config, you will be insurred what will happen if imponeer/composer-nodejs-installer for some unlikly reason will be deleted as composer package. In that case user will loose auto updates, but he still could use own package from last hardcoded nodejs version. 

I don't see much problems here but yeah I agree hardcoding package names is not a very good idea overall.

Reply:
I added posibility to specify any other package from where to get version with `package-for-node-version` and `package-for-yarn-version` config values.

For example, such composer json:
```yaml
{
  "name": "dummy/dummy",
  "description": "Dummy project to test if configuration works",
  "type": "project",
  "require": {
    "mariusbuescher/node-composer": "@dev",
    "something/yarn": "1.0.0",
  },
  "extra": {
        "mariusbuescher": {
            "node-composer": {
                "package-for-yarn-version": "something/yarn",
                "node-version": "6.11.0"            
            }
        }
    }
}
```
Will install yarn 1.0.0 and nodejs 6.11.0

Or if such composer.json:
```yaml
{
  "name": "dummy/dummy",
  "description": "Dummy project to test if configuration works",
  "type": "project",
  "require": {
    "mariusbuescher/node-composer": "@dev",
  },
  "extra": {
        "mariusbuescher": {
            "node-composer": {
                "package-for-yarn-version": "something/yarn",
                "node-version": "6.11.0"            
            }
        }
    }
}
```
Will install only node 6.11.0 because `something/yarn` is not required in project, so no yarn version could be resolved.

Right now is not fully dynamically (tries to find if imponeer packages are installed if not other specified) but I think this will solve most issues with `what if these imponeer packages will disapeer someday` but if they are installed your package will have just some new extra functionality that will not break old packages but if user decides to use it will add some extra value and some possibilities.